openreview/conference/matching.py

```diff
@@ -1,22 +1,91 @@
+'''
+A module containing tools for matching and and main Matching instance class
+'''
+
 from __future__ import division
-import openreview
 import csv
-from collections import defaultdict
+
+import openreview
+
+def _jaccard_similarity(list1, list2):
+    '''
+    Return a score, indicating the similarity between two lists
+    '''
+    set1 = set(list1)
+    set2 = set(list2)
+    intersection = set1.intersection(set2)
+    union = set1.union(set2)
+    return len(intersection) / len(union)
+
+def _get_profiles(client, ids_or_emails):
+    '''
+    Helper function that repeatedly queries for profiles, given IDs and emails.
+    Useful for getting more Profiles than the server will return by default (1000)
+    '''
+    ids = []
+    emails = []
+    for member in ids_or_emails:
+        if '~' in member:
+            ids.append(member)
+        else:
+            emails.append(member)
+
+    profiles = []
+    profile_by_email = {}
+
+    batch_size = 100
+    for i in range(0, len(ids), batch_size):
+        batch_ids = ids[i:i+batch_size]
+        batch_profiles = client.search_profiles(ids=batch_ids)
+        profiles.extend(batch_profiles)
+
+    for j in range(0, len(emails), batch_size):
+        batch_emails = emails[j:j+batch_size]
+        batch_profile_by_email = client.search_profiles(emails=batch_emails)
+        profile_by_email.update(batch_profile_by_email)
+
+    for email in emails:
+        profiles.append(profile_by_email.get(email, openreview.Profile(
+            id=email,
+            content={
+                'emails': [email],
+                'preferredEmail': email
+            })))
+
+    return profiles
 
 class Matching(object):
+    '''
+    Represents a Matching instance.
 
-    def __init__(self, conference):
+    Args:
+        `conference`: an openreview.Conference object
+        `match_group`: an openreview.Group object
+
+    '''
+    def __init__(self, conference, match_group):
         self.conference = conference
         self.client = conference.client
-
-    def _create_edge_invitation(self, id):
+        self.match_group = match_group
+
+    def _get_edge_invitation_id(self, edge_name):
+        '''
+        Returns a correctly formatted edge invitation ID for this Matching's match group
+        '''
+        return '{}/-/{}'.format(self.match_group.id, edge_name)
+
+    def _create_edge_invitation(self, edge_name):
+        '''
+        Creates an edge invitation given an edge name
+        e.g. "Affinity_Score"
+        '''
         invitation = openreview.Invitation(
-            id = id,
-            invitees = [self.conference.get_id()],
-            readers = [self.conference.get_id()],
-            writers = [self.conference.get_id()],
-            signatures = [self.conference.get_id()],
-            reply = {
+            id=self._get_edge_invitation_id(edge_name),
```

Comment:
Is this a pylint convention? I particularly prefer have spaces.

Reply:
I guess we are following these conventions: https://www.python.org/dev/peps/pep-0008/

I didn't know about them, I like to add spaces everywhere :/

Reply:
Yeah, I have just been following the default pylint settings. But I don't have a strong opinion about it, as long as we all use the same thing.

Reply:
Ok, let's follow the pylint conventions.